### ISSUE TITLE
Cannot assert that job has been enqueued with no wait

### ISSUE THREAD

**Issue by bcharna** [rspec/rspec-rails#1975]
I conditionally add a wait when creating a job like so:

```
options = {}
options[:wait_until] = @scheduled if @scheduled
MyJob.new(payload).enqueue(options)
```

I propose that `rspec-rails` gains support to be able to test that there is no wait when there should be no wait via something like
```
expect(MyJob).to have_been_enqueued.with(payload).with_no_wait
```
or
```
expect(MyJob).to have_been_enqueued.with(payload).at(:no_wait)
```

Thank you!

**Comment by JonRowe** [comment]
Will `.at(0)` not work?

**Comment by bcharna** [comment]
@JonRowe no, it doesn't. I wouldn't mind taking a deeper look at the code and seeing if I can implement myself, but would like to at least know the preferred API; `at(0)`, `at(:no_wait)`, etc?

**Comment by JonRowe** [comment]
`at(:no_wait)` seems reasonable, the code checks that its not nil but could then special case of that symbol and check for the appropriate internal value (which will probably be nil)

**Comment by bcharna** [comment]
Okay, sounds good. I will give it a go soon!

**Comment by bcharna** [comment]
Added a PR: https://github.com/rspec/rspec-rails/pull/1977

**Comment by samphippen** [comment]
This was fixed by #1977
